libimage/pull.go

```diff
@@ -592,6 +594,62 @@ func (r *Runtime) copySingleImageFromRegistry(ctx context.Context, imageName str
 		return nil
 	}
 
+	// Avoid running out of time when running inside a systemd unit by
+	// regularly increasing the timeout.
+	if socketPath, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
+		socketAddr := &net.UnixAddr{
+			Name: socketPath,
+			Net:  "unixgram",
+		}
+		conn, err := net.DialUnix(socketAddr.Net, nil, socketAddr)
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+
+		numExtensions := 10
+		extension := 30 * time.Second
+		timerFrequency := 25 * time.Second // Fire the timer at a higher frequency to avoid a race
+		timer := time.NewTimer(timerFrequency)
+		socketCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		defer timer.Stop()
+
+		logrus.Warnf("Pulling image %s inside systemd: setting pull timeout to %s", imageName, time.Duration(numExtensions)*extension)
```

Comment:
A warning seems a bit much? Having this logged for each systemd unit start adds more noise than I would like.
IMO it also does not really qualify as warning for me, more like info.

Reply:
Please wrestle this out with @rhatdan who requested it.  I am not married to the warning but see beauty in making this explicit.  It would otherwise be very hard to debug trying to understand why the configured timeout is being "ignored" by Podman.  An Info would be hidden in that case.

Reply:
I wonder if systemd already logs these notify messages, if it does we do not need to log it in podman. I will check it.

Reply:
Ok I do not see any hits of that happening in systemd unit logs. However I really think a warning sends the wrong message. A warning is something I as a user should be concerned about and likely need to act on. This here is working as intended so I think info level is the better choice.

Reply:
Info is not being displayed by default, so users will have a very hard time finding the log if being run via Quadlet.

I am not married to this warning but _some_ message/log that is being displayed by default and ends up in the journal and `systemctl status` sounds reasonable to me.

We can also not drop a log and try to make it very clear in the docs.  I _guess_ that only very advanced users will notice the potential timeout increase.

Reply:
@rhatdan WDYT?

Reply:
I would prefer info, but as you say they are not output automatically.  Should we handle this as we would progress and just write to Stderr a message that the pull has been extended?